File: my_timeboard.py
from datadog import initialize, api

options = {
    'api_key': 'f98954a3916a1ee5b81f204fffe4c652',
    'app_key': '14491168f3c77a6bad06c7d7487be3ae5719e52e'
}

initialize(**options)

title = "Kyle's TimeBoard"
description = "Solution's Engineer Code Challenge Timeboard"
graphs = [{
    "definition": {
        "events": [],
        "requests": [{
            "q": "avg:my_metric{host:ubuntu-xenial}",
            "type": "line",
        }],
        "viz": "timeseries"
    },
    "title": "Metric Scoped With Host",
}, {
    "definition": {
        "events": [],
        "requests": [{
         "q": "avg:postgresql.bgwriter.checkpoints_timed{host:ubuntu-xenial}.as_count()",
         "type": "line",
        }],
        "viz": "timeseries"
    },
    "title": "Anomoly monitor for PostgreSQL",
}, {
    "definition": {
        "events": [],
        "requests": [{
            "q": "avg:my_metric{*}.rollup(sum, 3600)",
            "type": "line",
        }],
        "viz": "timeseries"
    },
    "title": "Metric Rollup Point Sum",
}]

template_variables = [{
    "name": "host1",
    "prefix": "host",
    "default": "host:my-host"
}]

read_only = True

api.Timeboard.create(
    title=title,
    description=description,
    graphs=graphs,
    template_variables=template_variables,
    read_only=read_only
    )

File: myTimeboard.rb
require 'rubygems'
require 'dogapi'

api_key = '5ac8fd2b5d54f1dbe20ab426f983a251'
app_key = '764cdf93411341dd0dda6f061be2d3a7668e7ee1'

dog = Dogapi::Client.new(api_key, app_key)

title = 'Timeboard with my_metric, my_metric rollup, and anomolies on DB Color 2'
description = 'Test on My_Metric'
graphs = [
  {
      "definition" => {
          "events" => [],
          "requests" => [{
              "q" => "avg:my_metric{*}"
          }],
          "viz" => "timeseries"
      },
      "title" => "My_metric"
  },
  {
    "definition" => {
        "events" => [],
        "requests" => [{
            "q" => "anomalies(avg:mongodb.locks.collection.acquirecount.intent_exclusiveps{*}, 'basic', 3)",
            "style" => {
              "palette" => "purple"
            }
        }],
        "viz" => "timeseries"
    },
    "title" => "Anomalies on DB"
  },
  {
    "definition" => {
        "events" => [],
        "requests" => [{
            "q" => "sum:my_metric{*}.rollup(avg, 3600)"
        }],
        "viz" => "timeseries"
    },
    "title" => "Sum of my_metric 1hr Rollup"
  }
]
template_variables = [{
    "name" => "host1",
    "prefix" => "host",
    "default" => "host:my-host"
}]
dog.create_dashboard(title, description, graphs, template_variables)

# "q" => "avg:my_metric{*}"
# "q" => "sum:my_metric{*}.rollup(avg, 3600)"
